airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunner.kt

```diff
@@ -240,7 +250,11 @@ internal constructor(
                         .withConnectionStatus(
                             AirbyteConnectionStatus()
                                 .withStatus(AirbyteConnectionStatus.Status.FAILED)
-                                .withMessage(displayMessage)
+                                .withMessage(
+                                    ConnectorExceptionUtil.getDisplayMessage(
+                                        rootConfigErrorThrowable
```

Comment:
If errors are config error, and command is CHECK, we will not go into this special handling logic which might be a problem

Also is it true that errors in CHECK are always config errors?

Reply:
Ah you're right. Good catch. Let me try to fix this

Reply:
just reordered this

Reply:
could `rootConfigErrorThrowable` be null here if check encountered a non config error?

Reply:
See `getRootConfigError`. It returns the top-level exception in that case